src/components/Emotes.tsx

```diff
@@ -0,0 +1,376 @@
+    reloadEmotes();
+    forceUpdate();
+  }
+
+  const [copyButtonTitle, setCopyButtonTitle] = React.useState<string>(
+    t('Copy Code')
+  );
+  function copyEmoteScrapeScript() {
+    navigator.clipboard.writeText(emoteScrapeScript);
+    setCopyButtonTitle(t('Code Copied!'));
+  }
+
+  const [importState, setImportState] = React.useState<string>('');
+  async function importEmotesFromClipboard() {
+    try {
+      setImportState('import started');
+      const promises = await fetchEmotes(
+        JSON.parse(await navigator.clipboard.readText())
+      );
+      let numFinished = 0;
+      const progressUpdate = (message: string) => {
+        setImportState(`[${numFinished}/${promises.length}] ${message}`);
+      };
+      progressUpdate('Started downloads!');
+      await Promise.all(
+        promises.map((p) =>
+          p.then((filePathWithExtension: string) => {
+            numFinished += 1;
+            progressUpdate(filePathWithExtension);
+            return null;
+          })
+        )
+      );
+      progressUpdate('Done!');
+    } catch (err) {
+      setImportState(`error: ${err}`);
+      throw err;
+    }
+    reloadEmotesAndUpdate();
+  }
+
+  const element = (
+    <MuiThemeProvider theme={theme}>
+      <div className={classes.root}>
+        <div className={classes.content}>
+          <h2>Emote Tools</h2>
+
+          <Button
+            id="open-emote-directory"
+            variant="contained"
+            className={classes.button}
+            color="primary"
+            onClick={openEmoteDirectory}
+          >
+            {t('Open Emote Directory')}
+          </Button>
+          <Button
+            id="reload-emotes"
+            variant="contained"
+            className={classes.button}
+            color="primary"
+            onClick={reloadEmotesAndUpdate}
+          >
+            {t('Reload emotes')}
+          </Button>
+
+          <h2>Importing emotes</h2>
+          <div>
+            You can import/manage emotes manually using the buttons above and
+            placing images into that directory.
+          </div>
+          <div>
+            To import existing Twitch/BTTV/FFZ emotes you can do the following:
+            <ol>
+              <li>
+                {' '}
+                Install BTTV in your browser if you haven&rsquo;t already.{' '}
+              </li>
+              <li> Open your twitch channel page with chat. </li>
+              <li>
+                Open the browser console: Browser menu &gt; More Tools &gt; Web
+                Developer Tools &gt; &rdquo;Console&rdquo; tab
+              </li>
+              <li>
+                Note that pasting code into the browser console is not normal
+                and you should trust or verify the script. See
+                <a href="https://en.wikipedia.org/wiki/Self-XSS">Self-XSS</a>
+                for more info.
+              </li>
+              <li>
+                Click this button:
+                <Button
```

Comment:
The emote import process did not work for me. Upon clicking the import button, some empty folders of several twitch streams were created with no emotes. All emotes were missing URLs but had names in the JSON copied. The app froze during the import and had to be restarted.

Reply:
Added instructions and a check that emotes were loaded by BTTV.